### PR TITLE
Breakage report improvements

### PR DESCRIPTION
**Pull request by federicocappelli**
Task/Issue URL: https://app.asana.com/0/1205842942115003/1205692741026215/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2168
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1864
What kind of version bump will this require?: Major (Braking changes in privacy-dashboard)

**Description**:

The previous implementation of the `Report Broken Site` page was different between iOS and macOS, with this PR I have updated and aligned the PrivacyDashboardController so the web implementation of the `Report Broken Site` can be used in both iOS and macOS.
The changes are mostly around delegates and which User script is handled for each platform.
- Splitting the delegate into 3 new delegates: `PrivacyDashboardNavigationDelegate`, `PrivacyDashboardReportBrokenSiteDelegate`, `PrivacyDashboardControllerDelegate`

**Steps to test this PR**:

macOS: try `Privacy Dashboard` and the navigation to the web `Report Broken Site`, the functionality must remain the same.

iOS:
- Navigate to `Report Broken Site` from `Privacy Dashboard` shows the web page and all the features are correct
- Navigate to `Report Broken Site` from the app menu shows the web page and all the features are correct

**OS Testing**:

* [ ] iPhone 6s - iOS 15
* [ ] iOS 16
* [x] iPhone 15 - iOS 17
* [ ] macOS 10.15
* [ ] macOS 11
* [x] macOS 15

### CODE REVIEW

File: Sources/PrivacyDashboard/Model/TrackerInfo.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ import TrackerRadarKit
 import ContentBlocking
 
 public struct TrackerInfo: Encodable {
-
```
Comment (federicocappelli) [inline Sources/PrivacyDashboard/Model/TrackerInfo.swift:24]: all white space changes have been done by `swiftlint --fix`, the code indentation was not consistent across the project. I'll try to ignore the white spaces when pushing

Comment (federicocappelli) [reply Sources/PrivacyDashboard/Model/TrackerInfo.swift:24]: please use "Hide white spaces" here, after this fix this shouldn't be a problem anymore

**Review by tomasstrba — APPROVED** [review]
The code looks good to me 👍 I'll test and review the functionality as a part of iOS and macOS PRs.

Just one note: I am not sure what the settings of your dev environment are but this PR is hard to follow because of too much whitespace removals.

**Comment by federicocappelli** [comment]
@tomasstrba Thanks! I discussed the white spaces issue with Bartek, it's weird, I have the default xcode settings and I run swiftlint --fix, on the files with a lot of changes I re-did ctrl+i in xcode so they are now completely standard, maybe they had a weird indentation before?
